aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+    ///
+    /// This offers a migration path for an existing account with a multi-ed25519 auth key (native multisig account).
+    /// In order to ensure a malicious module cannot obtain backdoor control over an existing account, a signed message
+    /// with a valid signature from the account's auth key is required.
+    public entry fun create_with_existing_account(
+        multisig_account: &signer,
+        owners: vector<address>,
+        signatures_required: u64,
+        account_scheme: u8,
+        account_public_key: vector<u8>,
+        create_multisig_account_signed_message: vector<u8>,
+    ) {
+        // Verify that the `MultisigAccountCreationMessage` has the right information and is signed by the account
+        // owner's key.
+        let proof_challenge = MultisigAccountCreationMessage {
+            chain_id: chain_id::get(),
+        };
+        account::verify_signed_message(
+            address_of(multisig_account),
+            account_scheme,
+            account_public_key,
+            create_multisig_account_signed_message,
+            proof_challenge,
+        );
+
+        create_with_owners_internal(
+            multisig_account,
+            owners,
+            signatures_required,
+            option::none<SignerCapability>(),
+        );
+    }
+
+    /// Creates a new multisig account and add the signer as a single owner.
+    public entry fun create(owner: &signer, signatures_required: u64) {
+        create_with_owners(owner, vector[], signatures_required);
+    }
+
+    /// Creates a new multisig account with the specified additional owner list and signatures required.
+    ///
+    /// @param additional_owners The owner account who calls this function cannot be in the additional_owners and there
+    /// cannot be any duplicate owners in the list.
+    /// @param signatures_require The number of signatures required to execute a transaction. Must be at least 1 and
+    /// at most the total number of owners.
+    public entry fun create_with_owners(owner: &signer, additional_owners: vector<address>, signatures_required: u64) {
+        let (multisig_account, multisig_signer_cap) = create_multisig_account(owner);
+        vector::push_back(&mut additional_owners, address_of(owner));
+        create_with_owners_internal(
+            &multisig_account,
+            additional_owners,
+            signatures_required,
+            option::some(multisig_signer_cap),
+        );
+    }
+
+    fun create_with_owners_internal(
+        multisig_account: &signer,
+        owners: vector<address>,
+        signatures_required: u64,
+        multisig_account_signer_cap: Option<SignerCapability>,
+    ) {
+        assert!(
+            signatures_required > 0 && signatures_required <= vector::length(&owners),
+            error::invalid_argument(EINVALID_SIGNATURES_REQUIRED),
+        );
+
+        validate_owners(&owners, address_of(multisig_account));
+        move_to(multisig_account, MultisigAccount {
```

Comment:
At the current implementation, there is no way to track owned multisig of a certain account. Are you expecting third party to implement the multisig registry, or shall we create an event handle for each owners?

Reply:
Added an OwnedMultisigAccounts resource that's stored and maintained in each owner's account.